src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -302,11 +314,152 @@ protected virtual Operation GetOperation(string operationName, Document document
             return ExecuteFieldsAsync(context, rootType, context.RootValue, fields, new string[0]);
         }
 
-        public Task<Dictionary<string, object>> ExecuteFieldsAsync(ExecutionContext context, IObjectGraphType rootType, object source, Dictionary<string, Field> fields, IEnumerable<string> path)
+        public async Task<IDictionary<string, object>> ExecuteFieldsAsync(ExecutionContext context, IObjectGraphType rootType, object source, Dictionary<string, Field> fields, IEnumerable<string> path)
+        {
+
+            var data = new ConcurrentDictionary<string, object>();
+            var externalTasks = new List<Task>();
+
+            foreach (var fieldCollection in fields)
+            {
+                var currentPath = path.Concat(new[] { fieldCollection.Key });
+
+                var field = fieldCollection.Value;
+                var fieldType = GetFieldDefinition(context.Document, context.Schema, rootType, field);
+
+                if (fieldType?.Resolver == null || !fieldType.Resolver.RunThreaded() || context.Operation.OperationType == OperationType.Mutation)
+                {
+                    await ExtractFieldAsync(context, rootType, source, field, fieldType, data, currentPath);
+                }
+                else
+                {
+                    var task = Task.Run(() => ExtractFieldAsync(context, rootType, source, field, fieldType, data, currentPath));
+
+                    externalTasks.Add(task);
+                }
+            }
+
+            if (externalTasks.Any())
```

Comment:
Faster to use a .Count > 0 I believe?

Reply:
Thanks, this would be faster since its a list and not an enumerable